functional_tests/aws/user/test_update_template.py

```diff
@@ -216,3 +234,358 @@ async def test_replace_max_size_inline_policy(self):
+            : max(len(cls.all_account_ids) // 2, 2)
+        ]
+        asyncio.run(cls.template.apply(IAMBIC_TEST_DETAILS.config.aws))
+        cls.aws_account_map = asyncio.run(
+            get_aws_account_map(IAMBIC_TEST_DETAILS.config.aws)
+        )
+        cls.included_account_map = {
+            account_id: cls.aws_account_map[account_id]
+            for account_id in cls.template.included_accounts
+        }
+        cls.iam_user_template_map = asyncio.run(
+            get_existing_template_map(
+                repo_dir=IAMBIC_TEST_DETAILS.template_dir_path,
+                template_type=AWS_IAM_USER_TEMPLATE_TYPE,
+                template_map=IAMBIC_TEST_DETAILS.config.aws.template_map,
+                nested=True,
+            )
+        )
+
+    async def asyncSetUp(self):
+        await self.refresh_template()  # Ensure template state is correct
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.template.deleted = True
+        asyncio.run(cls.template.apply(IAMBIC_TEST_DETAILS.config.aws))
+
+    def set_enable_iam_user_credentials(self, enable_iam_user_credentials: bool):
+        IAMBIC_TEST_DETAILS.config.aws.enable_iam_user_credentials = (
+            enable_iam_user_credentials
+        )
+        for account in IAMBIC_TEST_DETAILS.config.aws.accounts:
+            account.enable_iam_user_credentials = enable_iam_user_credentials
+
+        for account in self.included_account_map.values():
+            account.enable_iam_user_credentials = enable_iam_user_credentials
+
+        for account in self.aws_account_map.values():
+            account.enable_iam_user_credentials = enable_iam_user_credentials
+
+    async def refresh_template(self):
+        # Need to trigger an IAMbic import for the user.
+        # It's just the easiest way to try update the template
+        aws_account = list(self.included_account_map.values())[0]
+        templatized_user_name = templatize_resource(aws_account, self.user_name)
+        exe_message = ExecutionMessage(
+            execution_id=str(uuid.uuid4()), command=Command.IMPORT, provider_type="aws"
+        )
+        user_messages = [
+            UserMessageDetails(
+                account_id=aws_account.account_id,
+                user_name=templatized_user_name,
+                delete=False,
+            )
+        ]
+        await collect_aws_users(
+            exe_message,
+            IAMBIC_TEST_DETAILS.config.aws,
+            self.iam_user_template_map,
+            detect_messages=user_messages,
+        )
+        await generate_aws_user_templates(
+            exe_message,
+            IAMBIC_TEST_DETAILS.config.aws,
+            IAMBIC_TEST_DETAILS.template_dir_path,
+            self.iam_user_template_map,
+            detect_messages=user_messages,
+        )
+        self.template = AwsIamUserTemplate.load(self.template.file_path)
+
+    async def set_user_password(self):
+        user_password = (
```

Comment:
context, this is a functional test. but since it is generating password, we probably should use the following

```
import secrets
user_password = secrets.token_hex(nbytes=32)
```

the context is random was called out to not used for security purpose:

Warning The pseudo-random generators of this module should not be used for security purposes. For security or cryptographic uses, see the [secrets](https://docs.python.org/3/library/secrets.html#module-secrets) module.

https://docs.python.org/3/library/random.html

secrets.token_hex([nbytes=None])
Return a random text string, in hexadecimal. The string has nbytes random bytes, each byte converted to two hex digits. If nbytes is None or not supplied, a reasonable default is used.